### PR TITLE
Fix segfault in al_draw_ribbon() when num_segments>128

### LINKED ISSUE #1157: segfault when trying to draw ribbons with more than 128 points
`ALLEGRO_VERTEX_CACHE_SIZE` is fixed, and small.

It would be nice to be able to draw arbitrarily long ribbons, but I could split them on my side.
Anyway, it should not segfault on Allegro side.

The simplest solution is to include a simple check at the beginning of `al_draw_ribbon()`:

    if (num_segments*2 > ALLEGRO_VERTEX_CACHE_SIZE) {
        fprintf(stderr,
            "Allegro is refusing to draw a ribbon with more than %d points.\n",
            ALLEGRO_VERTEX_CACHE_SIZE/2);
        return;
    }

### PR DESCRIPTION
**Pull request by bart9h**
Fixes #1157

### CODE REVIEW

**Comment by pedro-w** [comment]
I don't think this is right because, if thickness is zero, you can draw up
to ALLEGRO_VERTEX_CACHE_SIZE segments (agreed it's only
ALLEGRO_VERTEX_CACHE_SIZE/2 if thickness > 0)

File: addons/primitives/high_primitives.c [hunk 1]
```diff
@@ -1248,6 +1248,11 @@ void al_draw_ribbon(const float *points, int points_stride, ALLEGRO_COLOR color,
    LOCAL_VERTEX_CACHE;
    int ii;
 
+   if (num_segments*2 > ALLEGRO_VERTEX_CACHE_SIZE) {
+      fprintf(stderr, "Ribbon has too many segments.\n");
```
Comment (pedro-w) [inline addons/primitives/high_primitives.c:1252]: Probably should use `ALLEGRO_ERROR` here?

**Comment by bart9h** [comment]
Now I don't know if I should have left that \n or not..

**Comment by SiegeLord** [comment]
Thanks!

**Pull request merged**